server/issue.go

```diff
@@ -38,6 +40,104 @@ func makePost(userID, channelID, message string) *model.Post {
 	}
 }
 
+func (p *Plugin) httpOAuthConnect(w http.ResponseWriter, r *http.Request, instanceID types.ID) (int, error) {
+	code := r.URL.Query().Get("code")
+	userID := r.URL.Query().Get("state")
+
+	conf := p.getConfig()
+	jsonData, _ := json.Marshal(map[string]string{
+		"grant_type":    "authorization_code",
+		"client_id":     conf.JiraAuthAppClientID,
+		"client_secret": conf.JiraAuthAppClientSecret,
+		"code":          code,
+		"redirect_uri":  fmt.Sprintf("%s/oauth/connect", p.GetPluginURL()),
+	})
+
+	request, err := http.NewRequest(
+		"POST",
+		"https://auth.atlassian.com/oauth/token",
+		bytes.NewBuffer(jsonData),
+	)
+	if err != nil {
+		return respondErr(w, http.StatusBadRequest,
+			errors.Wrap(err, "Error creating get token request onauth2"))
+	}
+
+	request.Header.Set("Content-type", "application/json; charset=UTF-8")
+
+	client := &http.Client{}
+	response, err := client.Do(request)
+	if err != nil {
+		return respondErr(w, http.StatusBadRequest,
+			errors.Wrap(err, "Error requesting token with Bearer"))
+	}
+	defer response.Body.Close()
+
+	body, _ := ioutil.ReadAll(response.Body)
+
+	tokenKey := fmt.Sprintf("oauthcloud_token_%s", userID)
+	token := &OAuth2Token{}
+	err = json.Unmarshal(body, token)
+	if err != nil {
+		return respondErr(w, http.StatusBadRequest,
+			errors.Wrap(err, "No access token present when getting token from JIRA"))
+	}
+	p.client.KV.Set(tokenKey, token.AccessToken)
```

Comment:
We'll want to store the whole token rather than just the access token, so we can access the expiry time and refresh token

Reply:
Yes! In the new version I store all of it, but I haven't finished it yet, thank you for the comment!